django/urls/resolvers.py

```diff
@@ -277,47 +420,58 @@ def _populate(self):
         namespaces = {}
         apps = {}
         language_code = get_language()
-        for pattern in reversed(self.url_patterns):
-            if isinstance(pattern, RegexURLPattern):
-                self._callback_strs.add(pattern.lookup_str)
-            p_pattern = pattern.regex.pattern
-            if p_pattern.startswith('^'):
-                p_pattern = p_pattern[1:]
-            if isinstance(pattern, RegexURLResolver):
-                if pattern.namespace:
-                    namespaces[pattern.namespace] = (p_pattern, pattern)
-                    if pattern.app_name:
-                        apps.setdefault(pattern.app_name, []).append(pattern.namespace)
-                else:
-                    parent_pat = pattern.regex.pattern
-                    for name in pattern.reverse_dict:
-                        for matches, pat, defaults in pattern.reverse_dict.getlist(name):
-                            new_matches = normalize(parent_pat + pat)
-                            lookups.appendlist(
-                                name,
-                                (
-                                    new_matches,
-                                    p_pattern + pat,
-                                    dict(defaults, **pattern.default_kwargs),
+        try:
+            for url_pattern in reversed(self.url_patterns):
+                if not isinstance(url_pattern, BaseURL):
+                    msg = (
```

Comment:
Yes, it's redundant to have both.

To be honest I'm unsure about the type check here. Perhaps at some point someone will ask to support duck-typing...

In the current state of the path, this is also the only reason why the `BaseURL` class exists — it provides no behavior.

Perhaps removing this and the base class is the best solution? I don't know how confusing the stack trace looks if there's this kind of error in the URLconf.

Reply:
@sjoerdjob, do you remember why you added this? The system check still looks useful, although it's only working if the mistake is in the ROOT_URLCONF. The mistake in an included URL patterns now gives:
```
  File "/home/tim/code/django/django/urls/conf.py", line 49, in include
    if isinstance(url_pattern.pattern, LocalePrefixPattern):
AttributeError: 'tuple' object has no attribute 'pattern'
```
I'm going to fix that and remove this, I think. The test for this message uses `resolve('/')` but I don't think an invalid pattern would get past the system check in the first place, so the check seems unneeded.

Reply:
I think it had to do with `resolve` suddenly started raising errors during a refactoring. Not sure if I went down the rabbit hole that led me to make this change or not. If it can be gone, so much the better.